### PR TITLE
Create graph

### PR DESCRIPTION
**Pull request by Christian-B**
Builds on https://github.com/SpiNNakerManchester/PACMAN/pull/449

An ApplicationGraph is now created during the Writer.setup or mock call. So except for before sim.setup (or unittest_setup())
there is always a graph so no need to call create graph or set_(runtime)_ graph

The Graph Class (or at least what was actually needed) has been merged into ApplicationGraph.
With the following changes
- The Graph no longer has a label so all graph label params removed
- You can no longer set or get constraints on graph (Was totally unused)
- get_edges_ending_at_vertex method removed (no known use)
- add_vertices and add_edges methods removed as only used in tests
- Vertex are now only held as the Values in a dict.
    (This has a minor cost that you can not iterate vertices and add at the same time)
- The allowed Vertex and Edge type to add are now harded coded.

As the Graph is never cloned the clone method has been removed.

As the Graph is never given to the use the ApplicationGraphView is no longer need

Must be done after:
https://github.com/SpiNNakerManchester/SpiNNUtils/pull/184

Must be done at the same time as:
https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/961
https://github.com/SpiNNakerManchester/sPyNNaker/pull/1202
https://github.com/SpiNNakerManchester/SpiNNakerGraphFrontEnd/pull/217

tested by
http://apollo.cs.man.ac.uk:8080/blue/organizations/jenkins/Integration%20Tests/detail/create_graph/3/pipeline

Documented by
https://github.com/SpiNNakerManchester/SpiNNakerManchester.github.io/pull/42

### CODE REVIEW

File: pacman/data/pacman_data_view.py [hunk 1]
```diff
@@ -219,15 +210,12 @@ def get_n_vertices(cls):
...
-                cls.__pacman_data._graph = ApplicationGraph("Mocked")
-            else:
-                raise cls._exception("graph")
+            raise cls._exception("graph")
         return cls.__pacman_data._graph.n_vertices
 
     @classmethod
     def iterate_partitions(cls):
-        """ The partitions in the user application graph.
+        """ The partitions in the user application graphs an iterator
```
Comment (andrewgait) [inline pacman/data/pacman_data_view.py:218]: I think you mean "graph as an iterator"?

File: pacman/data/pacman_data_view.py [hunk 2]
```diff
@@ -271,69 +253,42 @@ def get_outgoing_edge_partitions_starting_at_vertex(cls, vertex):
...
         return sum(len(vertex.machine_vertices)
-                   for vertex in cls.get_runtime_graph().vertices)
+                   for vertex in cls.__pacman_data._graph.vertices)
 
     @classmethod
-    def get_runtime_machine_vertices(cls):
+    def iterate_machine_vertices(cls):
         """
-        Gets the Machine vertioces viua the application graph
+        Iterate the Machine vertices viua the application graph
```
Comment (andrewgait) [inline pacman/data/pacman_data_view.py:285]: "via"

File: pacman/model/graphs/application/application_graph.py [hunk 3]
```diff
@@ -18,44 +18,155 @@
...
+        :param ApplicationVertex vertex: The vertex to add
+        :raises PacmanInvalidParameterException:
+            If the vertex is not of a valid type
+        :raises PacmanConfigurationException:
+            If there is an attempt to add the same vertex more than once
+        """
+        if not isinstance(vertex, ApplicationVertex):
+            raise PacmanInvalidParameterException(
+                "vertex", str(vertex.__class__),
+                "Only ApplicationVertexs can be added")
```
Comment (andrewgait) [inline pacman/model/graphs/application/application_graph.py:58]: "Only an ApplicationVertex"

File: pacman/model/graphs/application/application_graph.py [hunk 4]
```diff
@@ -18,44 +18,155 @@
...
+                raise PacmanAlreadyExistsException("vertex", vertex.label)
+            vertex.set_label(vertex.label + self._label_postfix())
+        vertex.addedToGraph()
+        self._vertex_by_label[vertex.label] = vertex
+
+    @property
+    def vertices(self):
+        """ The vertices in the graph.
+
+        :rtype: iterable(AbstractVertex)
```
Comment (andrewgait) [inline pacman/model/graphs/application/application_graph.py:73]: Perhaps say that this is a dict rather than just an iterable?

**Review by andrewgait — APPROVED** [review]
This looks fine to me, just a few minor edits suggested

File: pacman/model/graphs/application/application_graph.py (hunk 4 above)
Comment (Christian-B) [reply pacman/model/graphs/application/application_graph.py:73]: The call returns only the values(0 of the dict not the whole dict. So iterable is correct

**Pull request merged**